server/services/generateReportService.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+//@ts-ignore
+import { Legacy } from 'kibana';
+import { Server, Request, ResponseToolkit } from 'hapi';
+import { RequestParams } from '@elastic/elasticsearch';
+import puppeteer from 'puppeteer';
+import imagesToPdf from 'images-to-pdf';
+import fs from 'fs';
+import { v1 as uuidv1 } from 'uuid';
+import { ServerResponse } from '../models/types';
+import { CLUSTER } from '../utils/constants';
+
+type ElasticsearchPlugin = Legacy.Plugins.elasticsearch.Plugin;
+
+export default class GenerateReportService {
+  esDriver: ElasticsearchPlugin;
+
+  constructor(esDriver: ElasticsearchPlugin) {
+    this.esDriver = esDriver;
+  }
+
+  report = async (req: Request, h: ResponseToolkit): Promise<any> => {
+    try {
+      const {
+        url,
+        itemName,
+        reportFormat,
+        windowWidth = 1200,
+        windowLength = 800,
+      } = req.payload as {
+        url: string;
+        itemName: string;
+        reportFormat: string;
+        windowWidth: number;
+        windowLength: number;
+      };
+
+      if (reportFormat === 'png') {
+        const { fileName } = await generatePNG(
+          url,
+          itemName,
+          windowWidth,
+          windowLength
+        );
+
+        //@ts-ignore
+        return h.file(fileName + '.png', { mode: 'attachment' });
+      } else if (reportFormat === 'pdf') {
+        const { fileName } = await generatePDF(
+          url,
+          itemName,
+          windowWidth,
+          windowLength
+        );
+        //@ts-ignore
+        return h.file(fileName + '.pdf', { mode: 'attachment' });
+      }
+
+      return { message: 'no support for such format: ' + reportFormat };
+    } catch (err) {
+      console.error('Reporting-Generate-PDF', err);
```

Comment:
Use logger instead of `console.log`

Reply:
Any logging library that you suggest?

Reply:
I also looked into AD implementation, do you think it's good to do the same thing as AD. https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/3ce118c7c2c49bcb859f84beaf763a36bf2ef00e/index.js#L60

Reply:
There is already logging with Kibana you just need to pass it over and use it.